src/main/java/com/google/api/codegen/transformer/py/PythonModelTypeNameConverter.java

```diff
@@ -0,0 +1,233 @@
+
+  @Override
+  public TypeName getTypeName(ProtoElement elem) {
+    if (elem instanceof EnumType) {
+      List<String> path = new LinkedList<>();
+      path.add(elem.getSimpleName());
+      for (ProtoElement elt = elem.getParent(); elt.getParent() != null; elt = elt.getParent()) {
+        path.add(0, elt.getSimpleName());
+      }
+      path.add(0, "enums");
+      String shortName = Joiner.on(".").join(path);
+      return getTypeNameInImplicitPackage(shortName);
+    }
+
+    String packageName = protoPackageToPythonPackage(elem.getFile().getProto().getPackage());
+    String filename = getPbFileName(elem.getFile().getSimpleName());
+    return typeNameConverter.getTypeName(packageName + "." + filename + "." + elem.getSimpleName());
+  }
+
+  @Override
+  public TypedValue getEnumValue(TypeRef type, EnumValue value) {
+    return TypedValue.create(getTypeName(type), "%s." + value.getSimpleName());
+  }
+
+  @Override
+  public TypedValue getZeroValue(TypeRef type) {
+    // Don't call getTypeName; we don't need to import these.
+    if (type.isMap()) {
+      return TypedValue.create(new TypeName("dict"), "{}");
+    }
+    if (type.isRepeated()) {
+      return TypedValue.create(new TypeName("list"), "[]");
+    }
+    if (PRIMITIVE_ZERO_VALUE.containsKey(type.getKind())) {
+      return TypedValue.create(getTypeName(type), PRIMITIVE_ZERO_VALUE.get(type.getKind()));
+    }
+    if (type.isMessage()) {
+      return TypedValue.create(getTypeName(type), "%s()");
+    }
+    if (type.isEnum()) {
+      return getEnumValue(type, type.getEnumType().getValues().get(0));
+    }
+    return TypedValue.create(new TypeName(""), "None");
+  }
+
+  @Override
+  public String renderPrimitiveValue(TypeRef type, String value) {
+    Type primitiveType = type.getKind();
+    if (!PRIMITIVE_TYPE_MAP.containsKey(primitiveType)) {
+      throw new IllegalArgumentException(
+          "Initial values are only supported for primitive types, got type "
+              + type
+              + ", with value "
+              + value);
+    }
+    switch (primitiveType) {
+      case TYPE_BOOL:
+        return Name.from(value.toLowerCase()).toUpperCamel();
+      case TYPE_STRING:
+      case TYPE_BYTES:
+        return "'" + value + "'";
+      default:
+        // Types that do not need to be modified (e.g. TYPE_INT32) are handled
+        // here
+        return value;
+    }
+  }
+
+  @Override
+  public TypeName getTypeNameForTypedResourceName(
+      FieldConfig fieldConfig, String typedResourceShortName) {
+    throw new UnsupportedOperationException(
+        "getTypeNameForTypedResourceName not supported by Python");
+  }
+
+  @Override
+  public TypeName getTypeNameForResourceNameElementType(
+      FieldConfig fieldConfig, String typedResourceShortName) {
+    throw new UnsupportedOperationException(
+        "getTypeNameForResourceNameElementType not supported by Python");
+  }
+
+  private String protoPackageToPythonPackage(String protoPackage) {
```

Comment:
IIUC this file duplicates some logic in `PythonImportHandler`. Is there a way to avoid the duplication? (Or if PythonImportHandler is going away with full MVVM, add a TODO?)